frontends/web/src/routes/account/account.tsx

```diff
@@ -49,324 +49,254 @@ const WithCoinTypeInfo = [
 
-  private onStatusChanged() {
-    const code = this.props.code;
-    if (!code) {
+  const onStatusChanged = useCallback(() => {
+    const currentCode = code;
+    if (!currentCode) {
       return;
     }
-    accountApi.getStatus(code).then(status => {
-      if (this.props.code !== code) {
+    accountApi.getStatus(currentCode).then(status => {
+      if (currentCode !== code) {
         // Results came in after the account was switched. Ignore.
         return;
       }
       if (!status.disabled) {
         if (!status.synced) {
-          accountApi.init(code).catch(console.error);
-        } else {
-          accountApi.getInfo(code)().then(accountInfo => {
-            if (this.props.code !== code) {
-              // Results came in after the account was switched. Ignore.
-              return;
-            }
-            this.setState({ accountInfo });
-          })
-            .catch(console.error);
+          accountApi.init(currentCode).catch(console.error);
         }
       }
-      this.setState({ status }, this.onAccountChanged);
+      setStatus(status);
     })
       .catch(console.error);
-  }
+  }, [code]);
 
-  private onAccountChanged = () => {
-    const status = this.state.status;
-    if (!this.props.code || status === undefined || status.fatalError) {
-      return;
-    }
-    if (status.synced && status.offlineError === null) {
-      const expectedCode = this.props.code;
-      Promise.all([
-        accountApi.getBalance(this.props.code).then(balance => {
-          if (this.props.code !== expectedCode) {
-            // Results came in after the account was switched. Ignore.
-            return;
-          }
-          this.setState({ balance });
-        }),
-        accountApi.getTransactionList(this.props.code).then(transactions => {
-          if (this.props.code !== expectedCode) {
-            // Results came in after the account was switched. Ignore.
-            return;
-          }
-          this.setState({ transactions });
-        })
-      ])
-        .catch(console.error);
-    } else {
-      this.setState({
-        balance: undefined,
-        transactions: undefined,
-      });
+  useEffect(onAccountChanged, [onAccountChanged, status]);
+
+  const subscriptions = useRef<UnsubscribeList>([]);
+  useEffect(() => {
+    unsubscribe(subscriptions.current);
+    subscriptions.current.push(
+      syncAddressesCount(code, (givenCode, addressesSynced) => {
+        if (givenCode === code) {
+          setSyncedAddressesCount(addressesSynced);
+        }
+      }),
+      statusChanged(code, () => onStatusChanged()),
+      syncdone(code, () => onAccountChanged()),
+    );
+
+    if (status === undefined) {
+      onStatusChanged();
     }
-  };
 
-  private export = () => {
-    if (this.state.status === undefined || this.state.status.fatalError) {
+    const unsubscribeList = subscriptions.current;
+    return () => unsubscribe(unsubscribeList); // TODO: test if subscriptions.current works
```

Comment:
also need to set `subscriptions.current = []` after calling `unsubscribe()`? otherwise the subscription lists grows everytime this effect is called?